base/terminology.bal

```diff
@@ -46,31 +48,40 @@ public class InMemoryTerminologyLoader {
     # load terminology
     # + return - Terminology populated
     public function load() returns Terminology|FHIRError {
-        ValueSet[] valueSetArray = [];
-        foreach json jValueSet in self.valueSets {
-            do {
-                ValueSet valueSet = check jValueSet.cloneWithType();
-                valueSetArray.push(valueSet);
-            } on fail error e {
-                return createFHIRError("Error occurred while type casting json value set to ValueSet type", ERROR,
-                                                                    PROCESSING, diagnostic = e.message(), cause = e);
+
+        map<CodeSystem> codeSystemMap = {};
+        foreach json jCodeSystem in self.codeSystems {
+            CodeSystem|error c = jCodeSystem.cloneWithType(CodeSystem);
+            if c is error {
+                FHIRError fHIRError = createFHIRError("Error occurred while type casting json code system to CodeSystem type", ERROR,
+                                                                        PROCESSING, diagnostic = c.message(), cause = c);
+                log:printError(fHIRError.toBalString());
+            } else {
+                string url = <string>c.url;
+                string version = <string>c.version;
+                string key = string `${url}|${version}`;
+                codeSystemMap[key] = c;
```

Comment:
This code block seems used in multiple places. Shall we extract that into function?

Reply:
I can't extract that into function because it refers to different record types but I reduced the line of codes by reducing redundant logics.